lms/djangoapps/course_goals/urls.py

```diff
@@ -6,11 +6,12 @@
 from django.conf.urls import include, url
 from rest_framework import routers
 
-from .views import CourseGoalViewSet
+from .views import CourseGoalViewSet, PopulateUserActivity
 
 router = routers.DefaultRouter()
 router.register(r'course_goals', CourseGoalViewSet, basename='course_goal')
 
 urlpatterns = [
     url(r'^v0/', include((router.urls, "api"), namespace='v0')),
+    url(r'^populate_user_activity$', PopulateUserActivity.as_view(), name='populate_user_activity'),
```

Comment:
good idea, i can add versioning
hm, i could name the endpoint something mobile specific - but I wonder if there is a use case in the future where something else could use it?
perhaps it being specific for the mobile apps should be noted in the comments?

well, originally i think the plan was just to hook into the backend to avoid the need for unnecessary extra requests.  
this seems feasible by hooking into one spot for web, but for mobile there's a ton of calls, which is why we've asked mobile to add a request on their end, since this implementation wouldn't be maintainable long term for mobile
if mobile is removed and there's a single hook in for web i think that would be ok

Reply:
I'm not just suggesting putting `mobile` in its name or adding a comment. I'm suggesting moving this endpoint to a djangoapp that is more suited - maybe `mobile_api`? Make this view a BFF for the mobile apps, rather than some generic versioned endpoint. (The lack of version was fine to my eyes - as long as its a BFF.)

Reply:
hm, well most mobile calls aren't going to the mobile_api, but sure i can make this specific to the mobile_api
you don't think there's possible future use cases where we'd need this for non-mobile reasons? (maybe if we need to back-populate some activity or something?)
though i do think most likely we won't need this except for mobile

Reply:
moved the api to the mobile_api django app

Reply:
> though i do think most likely we won't need this except for mobile

But I think (with my tacky "BFFs forever" T-shirt on), let's say the user profile MFE wanted to start recording user activity for goals (which doesn't make sense, but for the sake of argument). I'd argue that the `profile` MFE could add its own endpoint in its own BFF djangoapp. Or add a backend call to one of its existing endpoints.

The broader point here is that general purpose APIs, with arbitrary consumers, versioned (but never/rarely incremented) - we are moving away from those in general for this kind of work.

Consider the API Trichotomy: https://openedx.atlassian.net/wiki/spaces/AC/pages/790036554/API+Trichotomy+Proposal

We're building a feature-driven API - we only need to target specific clients and have no need for a support lifecycle implied by a version number, etc.

Reply:
ah, that doc is helpful context, good to know!